jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +67,97 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @param tempDirPath path to temporary jib local directory to use.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the jar file or if the directory for the
+   *     temporary directory path provided, doesn't exist.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath, @Nullable Path tempDirPath)
+      throws IOException {
+    Path localExplodedJarRoot;
+    if (tempDirPath == null) {
+      try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider()) {
+        localExplodedJarRoot = tempDirectoryProvider.newDirectory();
+      }
```

Comment:
Once a `Closeable` class is closed, it's permanently shutdown and not usable. So you should put the entire code that needs to use `localExplodedJarRoot` into the try-resource block.